From b3d63c6da0df32d283ce339b48f8fd5bacce6660 Mon Sep 17 00:00:00 2001
From: chandu078 <chandudyavanapelli03@gmail.com>
Date: Fri, 02 Oct 2026 11:42:28 +0000
Subject: [PATCH] kernel: Allow passing Bazel flags from BoardConfig

Forward TARGET_KERNEL_BAZEL_FLAGS to the kernel build so devices can
select their DTBO config.

Change-Id: I3f940d5d808c312330a3746ed91b9b7c92d28423
---

diff --git a/config/BoardConfigKernel.mk b/config/BoardConfigKernel.mk
index fdf5688..ca1a52a 100644
--- a/config/BoardConfigKernel.mk
+++ b/config/BoardConfigKernel.mk
@@ -45,6 +45,7 @@
 #                                          Defaults to empty
 #   TARGET_KERNEL_EXT_MODULES          = Optional, the external modules we are
 #                                          building. Defaults to empty
+#   TARGET_KERNEL_BAZEL_FLAGS          = Optional, additional Bazel flags for the kernel build
 #   TARGET_KERNEL_UNSAFE_DDK_HEADERS   = Specifies if bazel build should use unsafe headers for DDK
 #                                        modules, this defaults to empty and should only be set to
 #                                        true if no other choice.
@@ -139,7 +140,7 @@
 endif
 
 # Clear these first to prevent accidental poisoning from env
-KERNEL_BAZEL_FLAGS :=
+KERNEL_BAZEL_FLAGS := $(TARGET_KERNEL_BAZEL_FLAGS)
 KERNEL_MAKE_FLAGS :=
 
 ifeq ($(TARGET_KERNEL_UNSAFE_DDK_HEADERS),true)
